rocksdb_replicator/max_number_box.cpp

```diff
@@ -0,0 +1,102 @@
+/// Copyright 2016 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+//
+// @author bol (bol@pinterest.com)
+//
+
+#include "rocksdb_replicator/max_number_box.h"
+
+#include <algorithm>
+
+#include "folly/Likely.h"
+
+namespace replicator { namespace detail {
+
+void MaxNumberBox::post(const uint64_t num) {
+  std::vector<Waiter*> waiters_to_notify;
+  std::vector<Waiter*> waiters_to_wait;
+
+  {
+    std::unique_lock<std::mutex> lk(mtx_);
```

Comment:
in highqps condition, will mutex be less performant than other lock implementations?